### PR TITLE
RA-145

### PR DESCRIPTION
**Pull request by aoller**

### CODE REVIEW

File: VAS.Services/Controller/PlayListController.cs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+using VAS.Services.ViewModel;
+using VAS.Core.Store.Playlists;
+
+namespace VAS.Services.Controller
+{
+	public class PlayListController : IController
+	{
+		PlaylistCollectionVM viewModel;
+
+		public IPlayerController Player {
```
Comment (vguzmanp) [inline VAS.Services/Controller/PlayListController.cs]: I'm not sure if this should be a `IPlayerController` or a `PlayerVM`, ask @vmartos

File: VAS.Tests/Services/TestPlayListController.cs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+			string name = "name";
+			PlaylistCollectionVM playlistCollectionVM = new PlaylistCollectionVM ();
+			controller.SetViewModel (playlistCollectionVM);
+
+			mockDiaklogs.Setup (m => m.QueryMessage (It.IsAny<string> (), It.IsAny<string> (),
+				It.IsAny<string> (), It.IsAny<object> ())).Returns (AsyncHelpers.Return (name));
+
+			App.Current.EventsBroker.Publish<AddPlaylistElementEvent> (
+				new AddPlaylistElementEvent {
+					PlaylistElements = new System.Collections.Generic.List<IPlaylistElement> ()
```
Comment (vguzmanp) [inline VAS.Tests/Services/TestPlayListController.cs]: using

File: VAS.Tests/Services/TestPlayListController.cs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+			mockDiaklogs.Verify (guitoolkit => guitoolkit.QueryMessage (It.IsAny<string> (),
+				It.IsAny<string> (), It.IsAny<string> (), It.IsAny<object> ()), Times.Once ());
+
+			Assert.AreEqual (1, playlistCollectionVM.ViewModels.Count);
+			Assert.AreEqual (name, playlistCollectionVM.ViewModels [0].Name);
+		}
+
+	}
+}
```
Comment (vguzmanp) [inline VAS.Tests/Services/TestPlayListController.cs]: Test also adding to an existing playlist.

File: VAS.Tests/Services/TestPlayListController.cs [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+		{
+			string name = "name";
+			PlaylistCollectionVM playlistCollectionVM = new PlaylistCollectionVM ();
+			controller.SetViewModel (playlistCollectionVM);
+
+			mockDiaklogs.Setup (m => m.QueryMessage (It.IsAny<string> (), It.IsAny<string> (),
+				It.IsAny<string> (), It.IsAny<object> ())).Returns (AsyncHelpers.Return (name));
+
+			App.Current.EventsBroker.Publish<AddPlaylistElementEvent> (
+				new AddPlaylistElementEvent {
```
Comment (vguzmanp) [inline VAS.Tests/Services/TestPlayListController.cs]: Make the `Playlist = null` case explicit

File: VAS.Tests/Services/TestPlayListController.cs [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+			mockPlayerController = new Mock<IPlayerController> ();
+			mockDiaklogs = new Mock<IDialogs> ();
+		}
+
+		[SetUp ()]
+		public void Setup ()
+		{
+			mockGuiToolkit = new Mock<IGUIToolkit> ();
+			App.Current.GUIToolkit = mockGuiToolkit.Object;
+			mockDiaklogs = new Mock<IDialogs> ();
```
Comment (vguzmanp) [inline VAS.Tests/Services/TestPlayListController.cs]: the mocks are created both in the `TestFixtureSetup` and the `SetUp` methods

File: VAS.Tests/Services/TestPlayListController.cs [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+		}
+
+		[Test ()]
+		public void TestNewPlaylist ()
+		{
+			string name = "name";
+			PlaylistCollectionVM playlistCollectionVM = new PlaylistCollectionVM ();
+			controller.SetViewModel (playlistCollectionVM);
+
+			mockDiaklogs.Setup (m => m.QueryMessage (It.IsAny<string> (), It.IsAny<string> (),
```
Comment (vguzmanp) [inline VAS.Tests/Services/TestPlayListController.cs]: You could move all this to one of the `SetUp` methods

**Comment by vguzmanp** [comment]
Did you modify the PlaylistController (PlaylistManager) from LongoMatch?

File: VAS.Services/Controller/PlayListController.cs (hunk 1 above)
Comment (aoller) [inline VAS.Services/Controller/PlayListController.cs]: This should be a PlayerVM

File: VAS.Services/Controller/PlayListController.cs [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+			}
+		}
+
+		protected virtual Playlist HandleNewPlaylist (NewPlaylistEvent e)
+		{
+			string name = Catalog.GetString ("New playlist");
+			Playlist playlist = null;
+			bool done = false;
+			while (name != null && !done) {
+				name = App.Current.Dialogs.QueryMessage (Catalog.GetString ("Playlist name:"), null, name).Result;
```
Comment (aoller) [inline VAS.Services/Controller/PlayListController.cs]: As HandleNewPlaylist is not an async method, we need to use Result as synchronous approach.

**Comment by vguzmanp** [comment]
In order to keep the commit history clean, you should commit the fixes in separate commits, to squash them with the corresponding commit later.

If you fix everything in the same commit, it's impossible to squash with 3 different commits

File: VAS.Tests/Services/TestPlaylistController.cs [hunk 8]
```diff
@@ -79,7 +80,18 @@ public void TestNewPlaylist ()
 
 			Assert.AreEqual (1, playlistCollectionVM.ViewModels.Count);
 			Assert.AreEqual (name, playlistCollectionVM.ViewModels [0].Name);
-		}
 
+			App.Current.EventsBroker.Publish<AddPlaylistElementEvent> (
```
Comment (vguzmanp) [inline VAS.Tests/Services/TestPlaylistController.cs]: This is a different test

**Pull request merged**